@schematics/angular typescript dependency is still 2.6

### Bug Report or Feature Request
```
- [x] bug report
- [ ] feature request
```

### Area
```
- [ ] devkit
- [x] schematics
```

### Versions
`node --version` : v9.11.1
`npm --version` : v5.6.0
Linux Debian 9.4

### Repro steps
```
git clone git@github.com:noelmace/nx.git
cd nx
git checkout 9c1b1bf0aa2410c6ba32
yarn install
yarn build
```

### The log given by the failure
```
packages/bazel/src/utils/ast-utils.ts(32,38): error TS2345: Argument of type 'ts.SourceFile' is not assignable to parameter of type 'ts.SourceFile'.
  Types of property 'kind' are incompatible.
    Type 'ts.SyntaxKind.SourceFile' is not assignable to type 'ts.SyntaxKind.SourceFile'. Two different types with this name exist, but they are unrelated.
packages/bazel/src/utils/ast-utils.ts(206,38): error TS2345: Argument of type 'ts.SourceFile' is not assignable to parameter of type 'ts.SourceFile'.
packages/bazel/src/utils/ast-utils.ts(233,32): error TS2345: Argument of type 'ts.SourceFile' is not assignable to parameter of type 'ts.SourceFile'.
```

### Desired functionality
Upgrade @schematics/angular typescript dependency to 2.7.2 in order to play well with angular v6 and therefore @angular/cli v6.

### Mention any other details that might be useful

See https://github.com/noelmace/nx/commit/9c1b1bf0aa2410c6ba32c7787006c847bb8faf40 for the complete list of conflicting dependencies.

The code source corresponding to the failure's log is https://github.com/noelmace/nx/blob/9c1b1bf0aa2410c6ba32c7787006c847bb8faf40/packages/bazel/src/utils/ast-utils.ts

``yarn list --pattern typescript`` : 
```
yarn list v1.5.1
├─ @schematics/angular@0.5.4
│  └─ typescript@2.6.2
└─ typescript@2.7.2
Done in 1.11s.
```

In the mean time, I'm pretty sure their was a workaround for this kind of typings conflict, but I can't remember one. If someone knows and could share one, it would frees my mind (even if it's not necessary given that the upgrade is needed anyway).

#### investigations

So, I tried to look at this issue by myself, and found that : 
1. TypeScript ~2.6.2 was added to dependencies in v0.4.2 (by https://github.com/angular/devkit/commit/727e179052c264dd60357dba26e1bf362b5a76a6#diff-f1300208a045563d7d86dd174a32c14d) in order to resolve https://github.com/angular/devkit/issues/365
2. Angular-cli still rely on TS 2.6 in its devDependencies, but installing a local @schematics/angular with a TS dependency version set to 2.7 doesn't seems to breack any test
3. bumping TS dependency version in schematics/angular package.json doesn't  make any failure in devkit tests

So I'll make a minimal PR for that, and wait for feedbacks :slightly_smiling_face:

--- a/packages/schematics/angular/package.json
+++ b/packages/schematics/angular/package.json
@@ -14,6 +14,6 @@
   "dependencies": {
     "@angular-devkit/core": "0.0.0",
     "@angular-devkit/schematics": "0.0.0",
-    "typescript": "~2.6.2"
+    "typescript": ">=2.6.2 <2.8"
   }
 }

